Providers/Modules/Plugins/Common/plugin/heartbeat_lib.rb

```diff
@@ -0,0 +1,68 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+module HeartbeatModule
+  class LoggingBase
+    def log_error(text)
+    end
+  end
+
+  class RuntimeError < LoggingBase
+    def log_error(text)
+      $log.error "RuntimeError: #{text}"
+    end
+  end
+
+  class Heartbeat
+    require 'json'
+    require_relative 'oms_common'
+  
+    def initialize(error_handler)
+      @error_handler = error_handler
+    end
+
+    # match string of the form (1 or more non . chars)- followed by a . - (1 or more non . chars) - followed by anything
+    MAJOR_MINOR_VERSION_REGEX = /([^\.]+)\.([^\.]+).*/
+
+    def get_heartbeat_data_item(time, os_conf_file_path, agent_install_conf_file_path)
+      record = {}
+      record["Timestamp"] = OMS::Common.format_time(time)
+
+      fqdn = OMS::Common.get_fully_qualified_domain_name
+      if fqdn.nil?
+        record["Computer"] = OMS::Common.get_hostname
```

Comment:
copy the file from OMSAgent repro. It used to add "Device" property into the record. It changes to "Computer" now.